packages/react/src/components/cookieConsentCore/translations.js

```diff
@@ -0,0 +1,118 @@
+  }
+
+  // Normal strings as template strings, use like:
+  // index({ a: 1, b: 2 }, "${a} is smaller than ${b}")
+  // returns: 1 is smaller than 2
+  // https://stackoverflow.com/a/41540381
+  function index(obj, is, value) {
+    if (typeof is === 'string') {
+      is = is.split('.');
+    }
+    if (is.length === 1 && value !== undefined) {
+      obj[is[0]] = value;
+      return value;
+    }
+    if (is.length === 0) {
+      return obj;
+    }
+    return index(obj[is[0]], is.slice(1), value);
+  }
+
+  // Find translation based on key, fallback if not found
+  let translation = null;
+
+  const translationNotFound = !translations[key];
+  const translationIsAnArray = Array.isArray(translations[key]);
+  const translationIsObjectAndHasNoKeys =
+    typeof translations[key] === 'object' && !translationIsAnArray && Object.keys(translations[key]).length === 0;
+
+  // Tries to find translation in
+  //   given language,
+  //   fallback language or
+  //   first available language,
+  // in that order, returns translation key if not found
+
+  if (translationNotFound || translationIsObjectAndHasNoKeys) {
+    // If translation is missing, return key
+
+    translation = key;
+    // eslint-disable-next-line no-console
+    console.error(`Cookie consent: Missing translation key: ${key}, falling back to key as translation`);
+  } else if (typeof translations[key] === 'string' || typeof translations[key] === 'number') {
+    // Same translation for all languages.
+    translation = translations[key];
+  } else if (typeof translations[key] === 'object' && !translationIsAnArray) {
+    // Different translations for different languages.
+    if (translations[key][lang]) {
+      // Translation was found in given language, use it
+      translation = translations[key][lang];
+    } else if (translations[key][fallbackLang]) {
+      // Translation was not found in given language, but it was found in fallback language, use it
+      translation = translations[key][fallbackLang];
+      // eslint-disable-next-line no-console
+      console.error(`Cookie consent: Missing translation: ${key}:${lang}, using fallback language: ${fallbackLang}`);
+    } else {
+      // Translation was not found in given language or fallback language, use first available translation
+      const firstLang = Object.keys(translations[key])[0];
+      translation = translations[key][firstLang];
+      // eslint-disable-next-line no-console
+      console.error(
+        `Cookie consent: Missing primary and fallback translation: ${key}:${lang}/${fallbackLang}, using first known language: ${firstLang}`,
+      );
+    }
+  } else {
+    // Translation is not a string, number or object
+    throw new Error(`Cookie consent: Invalid translation: ${key}, should be string, number or object`);
+  }
+
+  if (translation) {
+    // Replace dollar strings in translation with corresponding data from parameters
+    return translation.replace(/\$\{.+?\}/g, (match) => {
```

Comment:
Umm, the translation function accepts either global strings or localised objects with translation key.

Sure, we could change the syntax, but we wanted to keep the json flexible so that you would not have to repeat the same string (for example localStorage) in every language if it's identical in all.

How strongly do you feel about changing the syntax?

Reply:
Hey, this `translations.replace` will fail if the value is a number since such a function is not available for number type. ping @Arkkimaagi

Reply:
What do you mean about "changed syntax"?